src/HasTranslatableSlug.php

```diff
@@ -120,4 +124,62 @@ protected function hasCustomSlugBeenUsed(): bool
 
         return $originalSlug !== $newSlug;
     }
+
+    public function resolveRouteBinding($value, $field = null): Model|null
+    {
+        $field = $field ?? $this->getRouteKeyName();
+        if ($field !== $this->getSlugOptions()->slugField) {
+            return parent::resolveRouteBinding($value, $field);
+        }
+        // Only some database types support json operations.
+        // If the database doesn't support it, null is returned as default method would do the same
+        try {
+            return $this
+                ->where("{$this->getSlugOptions()->slugField}->{$this->getLocale()}", $value)->first();
+        } catch (RuntimeException $exception) {
+            return null;
+        }
+    }
+
+    public function resolveSoftDeletableRouteBinding($value, $field = null): Model|null
```

Comment:
Isn't this nearly the same as `resolveRouteBinding`? Could we remove this duplication somehow?

Reply:
You are right. It is a lot of duplication. I used the original implementation and only updated what I needed to avoid any bug introduced by myself.